lib/console.js

```diff
@@ -235,18 +235,30 @@ Console.prototype.time = function time(label = 'default') {
 };
 
 Console.prototype.timeEnd = function timeEnd(label = 'default') {
+  const hasWarned = timeLogImpl('timeEnd', label);
+  if (!hasWarned) {
+    this._times.delete(label);
+  }
+};
+
+Console.prototype.timeLog = function timeLog(label, ...data) {
+  timeLogImpl('timeLog', label, data);
+};
+
+// Returns true if label was not found
+function timeLogImpl(name, label = 'default', data = []) {
   // Coerces everything other than Symbol to a string
   label = `${label}`;
   const time = this._times.get(label);
   if (!time) {
-    process.emitWarning(`No such label '${label}' for console.timeEnd()`);
-    return;
+    process.emitWarning(`No such label '${label}' for console.${name}()`);
+    return true;
   }
   const duration = process.hrtime(time);
   const ms = duration[0] * 1000 + duration[1] / 1e6;
-  this.log('%s: %sms', label, ms.toFixed(3));
-  this._times.delete(label);
-};
+  this.log('%s: %sms', label, ms.toFixed(3), ...data);
```

Comment:
Nit: I personally would prefer to just check if data is undefined by adding a `if / else` but as the function is not hot, that is just a non-blocking recommendation.

Reply:
I can do that :+1: